### PR TITLE
Rust support for stm32u5

### LINKED ISSUE #504: STM32U5
Cortex-M33, ultra lower power device. New out today. Let's get to it!

https://www.st.com/en/microcontrollers-microprocessors/stm32u5-series.html

### PR DESCRIPTION
**Pull request by Urhengulas**
This PR follows the steps in the README, section "adding-new-devices" in order to add support for the STM32U575 (and STM32U585).

First step of #504.

### CODE REVIEW

**Comment by Urhengulas** [comment]
Although I specified a stm32u575 and a stm32u585 feature in `makecrates.py`, the generated crate only has a `stm32u5` features.

`makecrates.py`:
```py
CRATE_DOC_FEATURES = {
    # ...
    "stm32u5": ["rt", "stm32u575", "stm32u585"],
}
```
`stm32u5/Cargo.toml`
```toml
[features]
default = ["rt"]
rt = ["cortex-m-rt/device"]
stm32u5 = []
```
Is this correct / okay? Or should I change something about it?

File: stm32_part_table.yaml [hunk 1]
```diff
@@ -747,3 +747,13 @@ stm32wb:
...
+
+stm32u5:
+  stm32u5x5:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32u5-series.html
+    rm: RM0456
+    rm_title: STM32U575 and STM32U585
+    rm_url: https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf
+    members:
+      - STM32U575
+      - STM32U575
```
Comment (newAM) [inline stm32_part_table.yaml:759]:
```suggestion
      - STM32U575
      - STM32U585
```

Maybe typo?  Might be related to only having a `stm32u5` feature.

Comment (Urhengulas) [reply stm32_part_table.yaml]: Thank you for spotting this 👍🏾

**Comment by Urhengulas** [comment]
I got it to generate the two device features `stm32u575` and `stm32u585` by duplicating the YAML patch file. I am not sure if this is correct, but hopefully someone knowledgeable can take a look. As far as I can see the PR is ready for review now 😄

**Comment by Urhengulas** [comment]
I am still unsure how I can test if the generated PAC actually works. I do have the board at home. Is there some basic example I can run to test this PR?

**Comment by newAM** [comment]
> I am still unsure how I can test if the generated PAC actually works. I do have the board at home. Is there some basic example I can run to test this PR?

Typically I do `make form -j 32` then try to compile the PAC in a binary, using `stm32u5.path = "/home/me/stm32-rs/stm32u5"` in `Cargo.toml`.

**Comment by burrbull** [comment]
Please, don't break history. Rebase. No merge in PR.

**Comment by Urhengulas** [comment]
> Please, don't break history. Rebase. No merge in PR.

Okay. I just used the "resolve conflict" button from GitHub.

**Comment by potentialdiffer** [comment]
Hello together, 
what is the status of this? @Urhengulas can you rebase your branch?
Kind regards!

**Comment by Urhengulas** [comment]
> Hello together, what is the status of this? @Urhengulas can you rebase your branch? Kind regards!



> Hello together, what is the status of this? @Urhengulas can you rebase your branch? Kind regards!

Just did

**Review by newAM — APPROVED** [review]
Everything builds, sha256 for the SVD matches with the latest from ST (`07a14ac0257ab7304f282186b4db346e853f14c96e5ae9f653d557e353ac63e7` for the 2023-01-19 release), looks good to me, thanks!

bors r+

**Pull request merged**